src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -124,10 +122,11 @@ private void ifUsernameAlreadyExistsThrowException(String username) {
   }
 
   private User findUserToActivateOrThrowException(UUID activationTokenUUID) {
-    UUID userUUID = tokenService.findUser(activationTokenUUID);
-    return userRepository.findByUserIdAndActivatedIsFalse(userUUID)
-            .orElseThrow(() -> new IllegalArgumentException("Token is invalid or user is already activate."));
-
+    User user = tokenService.findUserByActivationToken(activationTokenUUID);
+    if(user.isActivated()){
+      throw new IllegalArgumentException("User with UUID " + user.getUserId() + " is already activated.");
+    }
```

Comment:
Movámoslo a un método privado
```suggestion
    throwExceptionIfUserIsActivated(User user);
```
```
private void throwExceptionIfUserIsActivated(User user) {
    if (user.isActivated()) {
      throw new IllegalArgumentException("User with UUID " + user.getUserId() + " is already activated.");
    }
}
```